### ISSUE TITLE
Brush endIndex is resetting when chart data changes while Brush endIndex is set to 0.

### ISSUE THREAD

**Issue by tryonelove** [recharts/recharts#3074]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/recharts-issue-template-forked-il4xyi?file=/src/index.js)

### Steps to reproduce
1. Set brush traveller start/end index to 0.
2. Hover over chart

### What is expected?
Expected chart with a single bar, nothing should be happening. 

### What is actually happening?
Brush range state is reset to 0-MAX_INDEX instead of being the same (0-0).

| Environment | Info |
|---|---|
| Recharts | v2.1.6 |
| React | 17.0.1 |
| System | Windows 11 |
| Browser | Chrome |
---
[Bug](https://github.com/recharts/recharts/blob/6c692d3919b75c348b9d4a034aca011fb40f0202/src/chart/generateCategoricalChart.tsx#L538)

Basically. brush reset is happening because of the following line:
`(brushItem && brushItem.props && brushItem.props.endIndex) || (props.data && props.data.length - 1) || 0;`

Since endIndex is set to 0 and props.data.length - 1 > 0, the code is using props.data.length over endIndex, which is required at the moment. `brushItem.props.endIndex` should be explicitely checked for undefined / null.
